src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+                count = 2 if "concat" in config.link_tower_type else 1
+                self.dense = nn.Sequential(
+                    nn.Linear(self.hidden_size * count, self.hidden_size * count),
+                    nn.LayerNorm(self.hidden_size * count),
+                    nn.GELU(),
+                    nn.Linear(self.hidden_size * count, self.hidden_size),
+                )
+            self.LayerNorm = nn.LayerNorm(self.hidden_size)
+        elif config.link_tower_type in ["cross_attention", "cross_attention_ffn"]:
+            self.dense = BridgeTowerBertLinkLayer(tokenizer_config, self.link_tower_type)
+        else:
+            raise NotImplementedError(f"link_tower_type {config.link_tower_type} is not implemented")
+
+    def forward(self, hidden_states, cross_modal_hidden_states, attention_mask):
+        if self.link_tower_type == "add":
+            return self.LayerNorm(hidden_states + cross_modal_hidden_states)
+        elif self.link_tower_type in ["scaled_add", "scaled_add_tensor"]:
+            return self.LayerNorm(hidden_states * self.scaled_factor + cross_modal_hidden_states)
+        elif self.link_tower_type in ["interpolate", "interpolate_tensor"]:
+            return self.LayerNorm(hidden_states * (1 - self.beta) + cross_modal_hidden_states * self.beta)
+        elif self.link_tower_type == "gated_interpolate":
+            scaled_gate = self.gate(torch.cat([hidden_states, cross_modal_hidden_states], dim=-1))
+            return self.LayerNorm(hidden_states * (1 - scaled_gate) + cross_modal_hidden_states * scaled_gate)
+        elif self.link_tower_type in ["elementwise_product", "elementwise_product_residual"]:
+            if "residual" in self.link_tower_type:
+                return self.LayerNorm(cross_modal_hidden_states + hidden_states * cross_modal_hidden_states)
+            else:
+                return self.LayerNorm(hidden_states * cross_modal_hidden_states)
+        elif self.link_tower_type in ["linear_add_fpn"]:
+            return self.LayerNorm(self.dense1(hidden_states) + self.dense2(cross_modal_hidden_states))
+        elif self.link_tower_type in ["linear_concat", "linear_concat_residual", "mlp_concat", "mlp_concat_residual"]:
+            if "residual" in self.link_tower_type:
+                return self.LayerNorm(
+                    cross_modal_hidden_states
+                    + self.dense(torch.cat([hidden_states, cross_modal_hidden_states], dim=-1))
+                )
+            else:
+                return self.LayerNorm(self.dense(torch.cat([hidden_states, cross_modal_hidden_states], dim=-1)))
+        elif self.link_tower_type in ["linear_add", "linear_add_residual", "mlp_add", "mlp_add_residual"]:
+            if "residual" in self.link_tower_type:
+                return self.LayerNorm(
+                    cross_modal_hidden_states + self.dense(hidden_states + cross_modal_hidden_states)
+                )
+            else:
```

Comment:
Same comment as above!

Reply:
Cleaned up the code in the updated commits.